kernels/triton/inference/mamba/causal_1d_conv/causal_1d_conv/causal_1d_conv.py

```diff
@@ -328,3 +331,324 @@ def causal_conv1d_fn(
+            out,
+            # Matrix dimensions
+            batch,
+            dim,
+            seqlen,
+            state_len,
+            num_cache_lines,
+            # stride
+            x.stride(0),  # X (batch, dim, seqlen)
+            x.stride(1),
+            x.stride(2),
+            stride_w_dim,
+            stride_w_width,
+            conv_state.stride(0),
+            conv_state.stride(1),
+            conv_state.stride(2),
+            out.stride(0),
+            out.stride(1),
+            out.stride(2),
+            # others
+            pad_slot_id,
+            # META
+            HAS_BIAS=bias is not None,
+            KERNEL_WIDTH=width,
+            SILU_ACTIVATION=activation in ["silu", "swish"],
+            IS_CONTINUOUS_BATCHING=conv_state_indices is not None,
+            IS_CIRCULAR_BUFFER=cache_seqlens is not None,
+            NP2_STATELEN=triton.next_power_of_2(state_len),
+            USE_PAD_SLOT=pad_slot_id is not None,
+        )
+    if unsqueeze:
+        out = out.squeeze(-1)
+    return out
+
+
+def causal_conv1d_update_vllm(
+    x: torch.Tensor,
+    conv_state: torch.Tensor,
+    weight: torch.Tensor,
+    bias: Optional[torch.Tensor] = None,
+    activation: Optional[Literal["silu", "swish"]] = None,
+    cache_seqlens: Optional[torch.Tensor] = None,
+    conv_state_indices: Optional[torch.Tensor] = None,
+    pad_slot_id: int = PAD_SLOT_ID,
+):
+    """
+    x: (batch, dim) or (batch, dim, seqlen)
+        [shape=2: single token prediction]
+        [shape=3: multiple tokens prediction]
+    conv_state: (batch, dim, state_len), where state_len >= width - 1
+    weight: (dim, width)
+    bias: (dim,)
+    cache_seqlens: (batch,), dtype int32.
+        If not None, the conv_state is treated as a circular buffer.
+        The conv_state will be updated by copying x to the conv_state
+        starting at the index
+        @cache_seqlens % state_len.
+    conv_state_indices: (batch,), dtype int32
+        If not None, the conv_state is a larger tensor along the batch dim,
+        and we are selecting the batch coords specified by conv_state_indices.
+        Useful for a continuous batching scenario.
+    pad_slot_id: int
+            if cache_indices is passed, lets the kernel identify padded
+            entries that will not be processed,
+            for example: cache_indices = [pad_slot_id, 1 ,20 ,pad_slot_id]
+            in this case, the kernel will not process entries at
+            indices 0 and 3
+    out: (batch, dim) or (batch, dim, seqlen)
+    """
+    assert cache_seqlens is None
+    # TODO : adopt the strategy in vLLM that overwrite on 'x' directly, rather than creating a new tensor 'o'
+    o = causal_conv1d_update(
+        x,
+        conv_state,
+        weight,
+        bias=bias,
+        activation=activation,
+        cache_seqlens=cache_seqlens,
+        conv_state_indices=conv_state_indices,
+        pad_slot_id=pad_slot_id,
+    )
+    return o
```

Comment:
nit but not a fan of using o by itself... out or output etc. makes it more clear imo.